Client/Application/NavigationRouter.swift

```diff
@@ -173,7 +174,9 @@ enum NavigationPath {
         case .defaultBrowser(let path):
             NavigationPath.handleDefaultBrowser(path: path)
         case .referral(let code):
-            bvc.openBlankNewTabAndClaimReferral(code: code)
+            if Referrals.isEnabled {
```

Comment:
This code would have made sense in 6.0.3 to check for the compile flag deactivation, but now that this is removed entirely, we don't need the `if Referrals.isEnabled` here. 

In fact it's a feature that we want to "override" Goodall referral feature flag (which is behing Referrals.isEnabled) by an incoming deeplink.

Reply:
@ecotopian Sure I can remove it though seems a bit flaky, having a feature flag we want overridden by design.

By this convention the feature flag only controls sharing a code. But if we turn it off we can't prevent previously created codes/deep links to still be shared.

Reply:
Yes thanks, I agree it's tricky, but it is like this in order to role out gradually. We can still turn it off completely in Goodall (having none of the variants "control" or "test" - then redeeming will work still, you're right, but UI will not reflect it anymore and sharing won't be possible anymore either. So that's due to the trickiness of this feature that 2 clients are involved.

So basically "only" the assignment of "test" group is overriden, not the entire flag. And redemption itself is not behind the flag, but it's result won't be user-facing if toggled off. We could add a second level to the feature flag to make this more transparent. e.g. having something like `isEnabled` (flag is toggled on generally) and `isActive` (flag is toggled on generally **and** user is in test group) What do you think?

Reply:
My point is not merely transparency. Let's imagine the scenario when you share your code with someone via email. 

That email could live there indefinitely. We decide to turn off the feature for whatever reason, but some time later you still have that email lying around, so even if the sharing UI is hidden under the feature flag, you can still redeem this code.

But if you don't see a problem with that then I'm fine too 👍

Reply:
yeah I see your point, but that's fine for first iteration. The link will open the app, but just not work anymore for the case it's toggled off and that is expected. Happens with old links all the time 🤷

Reply:
Wouldn't the link open the app and try to claim it?